### DISCUSSION TITLE
Command failed with exit code 243: npm install directus --production --no-optional

### DISCUSSION THREAD

**Discussion by i-pasha** [directus/directus#5586] [Archive]
Heyo,

i have been trying to follow the guide https://docs.directus.io/guides/installation/cli/ 
Context: I want to create an instance with an aws s3 connection.
unfortunately i cant get through step 2 because the "npx create-directus-project" command does not want to work.
Here is the "crash"-log:

```
sudo npx create-directus-project boi

npx: installed 54 in 3.129s
⠏ Installing Directus(node:27728) UnhandledPromiseRejectionWarning: Error: Command failed with exit code 243: npm install directus --production --no-optional
    at makeError (/root/.npm/_npx/27728/lib/node_modules/create-directus-project/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/root/.npm/_npx/27728/lib/node_modules/create-directus-project/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Command.create (/root/.npm/_npx/27728/lib/node_modules/create-directus-project/lib/index.js:62:2)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:27728) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:27728) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Noticeable to mention is that the command is stuck in a "Installing Directus" loop and the following files are created:

```
extensions
package.json
uploads
```

I wish everyone who reads this a nice day and thank you in advance ^^
Alex

**Comment by rijkvanzanten** [comment]
Hi Alex! This sounds like an npm permissions issue to me. Why are you running `npm` using sudo?

**Reply by i-pasha** [reply]
I tried using with and without sudo. No difference i saw

**Reply by rijkvanzanten** [reply]
The error 243 is from your filesystem and means a permission denied in wherever npm is trying to install Directus 🤔

**Reply by i-pasha** [reply]
I am installing the package under /home/<user-id>/<Folder with 777 and chown -R> with / without sudo
I validated that "npm config get prefix" is returning  "/usr/local/"
I made sure that I ran "sudo chown -R $(whoami) ~/.npm"
But still the same error remains. Am I missing a crucial step here? Is the npm version important? Does npm need additional tools to work properly? Are there other folders that struggle with permission shenanigans?

**Reply by rijkvanzanten** [reply]
I have no idea! Permission issues in npm itself can vary wildly between platforms. Based on your folder structure, I'm assuming you're in some Linux distro? How did you install Node/npm in the first place? 

The `npx` step will try installing the create package globally, maybe that global package is going to the wrong place? Could you try installing manually so it doesn't have this global installation requirement? https://docs.directus.io/guides/installation/manual/
